src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/IndexStateManagementHistory.kt

```diff
@@ -0,0 +1,200 @@
+            rolloverHistoryIndex()
+            // schedule the next rollover for approx MAX_AGE later
+            scheduledRollover = threadPool.scheduleWithFixedDelay({ rolloverAndDeleteHistoryIndex() }, historyRolloverCheckPeriod, executorName())
+        } catch (e: Exception) {
+            // This should be run on cluster startup
+            logger.error("Error creating ISM history index.", e)
+        }
+    }
+
+    override fun offMaster() {
+        scheduledRollover?.cancel()
+    }
+
+    override fun executorName(): String {
+        return ThreadPool.Names.MANAGEMENT
+    }
+
+    private fun rescheduleRollover() {
+        if (clusterService.state().nodes.isLocalNodeElectedMaster) {
+            scheduledRollover?.cancel()
+            scheduledRollover = threadPool.scheduleWithFixedDelay({ rolloverAndDeleteHistoryIndex() }, historyRolloverCheckPeriod, executorName())
+        }
+    }
+
+    private fun rolloverAndDeleteHistoryIndex() {
+        if (historyEnabled) rolloverHistoryIndex()
+        deleteOldHistoryIndex()
+    }
+
+    private fun rolloverHistoryIndex(): Boolean {
+        if (!indexStateManagementIndices.indexStateManagementIndexHistoryExists()) {
+            return false
+        }
+
+        // We have to pass null for newIndexName in order to get Elastic to increment the index count.
+        val request = RolloverRequest(IndexStateManagementIndices.HISTORY_WRITE_INDEX_ALIAS, null)
+        request.createIndexRequest.index(IndexStateManagementIndices.HISTORY_INDEX_PATTERN)
+            .mapping(_DOC, indexStateManagementIndices.indexStateManagementHistoryMappings, XContentType.JSON)
+        request.addMaxIndexDocsCondition(historyMaxDocs)
+        request.addMaxIndexAgeCondition(historyMaxAge)
+        val response = client.admin().indices().rolloversIndex(request).actionGet()
+        if (!response.isRolledOver) {
+            logger.info("${IndexStateManagementIndices.HISTORY_WRITE_INDEX_ALIAS} not rolled over. Conditions were: ${response.conditionStatus}")
+        }
+        return response.isRolledOver
+    }
+
+    @Suppress("SpreadOperator")
+    private fun deleteOldHistoryIndex() {
+        val indexToDelete = mutableListOf<String>()
+
+        val clusterStateRequest = ClusterStateRequest()
+        clusterStateRequest.clear().indices(IndexStateManagementIndices.HISTORY_ALL).metaData(true)
+        clusterStateRequest.local(true)
+        val strictExpandIndicesOptions = IndicesOptions.strictExpand()
+        clusterStateRequest.indicesOptions(strictExpandIndicesOptions)
+
+        val clusterStateResponse = client.admin().cluster().state(clusterStateRequest).actionGet()
+
+        for (entry in clusterStateResponse.state.metaData.indices()) {
+            val indexMetaData = entry.value
+            val creationTime = indexMetaData.creationDate
+
+            logger.info("CreationTime $creationTime $historyRetentionPeriod.millis}")
```

Comment:
This is a lot of info logs, some clusters can have thousands of indices and this will spam logs.